commit 62cb31d0a074adf6bc2cb321cc55ea51d5f69f5d
Author: Cody Russell <cody@jhu.edu>
Date:   Fri Apr 17 14:59:40 2015 -0500

    Unset the first responder if the newly focused widget is not a GtkNSView.

diff --git a/gtk/gtknsview.c b/gtk/gtknsview.c
index 645231d..34c5166 100644
--- a/gtk/gtknsview.c
+++ b/gtk/gtknsview.c
@@ -551,7 +551,7 @@ gtk_ns_view_notify (GObject    *object,
 
       if (gtk_widget_has_focus (GTK_WIDGET (object)))
         [ns_window makeFirstResponder:ns_view->priv->view];
-      else if ([ns_window firstResponder] == ns_view->priv->view || !gtk_window_is_active (window))
+      else if (!GTK_IS_NS_VIEW (gtk_window_get_focus (window)) || !gtk_window_is_active (window))
         [ns_window makeFirstResponder:nil];
     }
 }
